149 - Restructure Shared Components Folders

nozomione opened the pull request:
## Issue Number

Closing #149

- The base PR was [#145](https://github.com/AlexsLemonade/refinebio-web/pull/145) (merged into `dev`) 
- The commit of this PR is 1f5d64f2e4ebe9c46fc51b2c711f9a6b4a3c9f91 and ends at c3c0fdbb57e6e5c784c9287613cb135d60643f85

## Purpose/Implementation Notes
Please see the latest UI [here](https://refinebio-web-git-nozomione-149-restructure-shared-9771b5-ccdl.vercel.app).

Updated the shard components folders and names and fixed the imports.

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist
- [x] Lint and unit tests pass locally with my changes

davidsmejia reviewed (changes requested):
Please remove commit  af05657267e125a56055793e0731368f97823f2d from this PR. This PR should only be the restructuring of the src/components folder. We should group the yarn/nextjs changes in a different PR. thank you

nozomione commented:
> Please remove commit [af05657](https://github.com/AlexsLemonade/refinebio-web/commit/af05657267e125a56055793e0731368f97823f2d) from this PR. This PR should only be the restructuring of the src/components folder. We should group the yarn/nextjs changes in a different PR. thank you


@davidsmejia  I've set yarn's latest stable version back to v1.xx and removed the af05657267e125a56055793e0731368f97823f2d change, pls take a look, thank you!

davidsmejia reviewed (changes requested):
Please remove `.yarnrc.yml` and revert the changes in `yarn.lock` as well. Again I would like to not have anything in this PR besides what was discussed in the ticket. Thanks

nozomione commented:
> Please remove `.yarnrc.yml` and revert the changes in `yarn.lock` as well. Again I would like to not have anything in this PR besides what was discussed in the ticket. Thanks

Opps, sorry for the yarnrc yml! I reverted back yarn.lock to exactly match the previously closed PR #145 version (I included PATCH and MINOR releases for packages but they're now removed. We should included these releases in the future PR since they are [backward compatible]( https://semver.org) and fix some 🐛 etc. Please take a look, thank you!

davidsmejia commented on src/components/shared/DataTable/DataTable.js:
```diff
@@ -11,16 +11,10 @@ import { useSticky } from 'react-table-sticky'
 import { Box, Text } from 'grommet'
 import { useIntersectObserver } from 'hooks/useIntersectObserver'
 import { BoxBlock } from 'components/shared/BoxBlock'
-import {
```
Please don't use relative imports here

nozomione replied:
In this case, we need to re-do the merged PR https://github.com/AlexsLemonade/refinebio-web/pull/145 as well. Then I'll create a new PR to adjust all the relative imports, please let me know!

nozomione replied:
🗒️  We'll use the following pattern for module imports:
- use the absolute path (`src` as the base url) for shared or page components
- use the relative path for nested components (child components that belong to that parent module and only used in it. e.g., for `ProcessingInformationCell` > `ModalContent`,  use `./ModalContent` since this modal content is specific to the processing information cell and not shared elsewhere)

davidsmejia reviewed (changes requested):
Please update the import paths to start from `components`.

davidsmejia reviewed (approved):
LGTM
